zipline/finance/execution.py

```diff
@@ -86,9 +86,17 @@ def __init__(self, limit_price, exchange=None):
 
         self.limit_price = limit_price
         self._exchange = exchange
+        self.asset = asset
 
     def get_limit_price(self, is_buy):
-        return asymmetric_round_price_to_penny(self.limit_price, is_buy)
+        if hasattr(self.asset, 'tick_size'):
```

Comment:
It looks like roughly the same block is repeated ~4 times. Can the pattern be abstracted?

Reply:
Would make sense to put `tick_size` on all asset types, with a default value? Any performance or domain modeling concerns?